app/tests/test_generic_api.py

```diff
@@ -39,95 +39,116 @@ class ApiPaginationTestCase(StacBaseTestCase):
+                                 response.json()['description'],
+                                 msg='Unexpected error message')
+
+                response = self.client.get(f"/{STAC_BASE_V}/{endpoint}?limit=test")
+                self.assertStatusCode(400, response)
+                self.assertEqual(['invalid limit query parameter: must be an integer'],
+                                 response.json()['description'],
+                                 msg='Unexpected error message')
+
+                response = self.client.get(f"/{STAC_BASE_V}/{endpoint}?limit=-1")
+                self.assertStatusCode(400, response)
+                self.assertEqual(['limit query parameter too small, must be in range 1..100'],
+                                 response.json()['description'],
+                                 msg='Unexpected error message')
+
+                response = self.client.get(f"/{STAC_BASE_V}/{endpoint}?limit=1000")
+                self.assertStatusCode(400, response)
+                self.assertEqual(['limit query parameter too big, must be in range 1..100'],
+                                 response.json()['description'],
+                                 msg='Unexpected error message')
 
     def test_pagination(self):
+        items = self.factory.create_item_samples(3, self.collections[0].model, db_create=True)
+        for endpoint in [
+            'collections',
+            f'collections/{self.collections[0]["name"]}/items',
+        ]:
+            with self.subTest(endpoint=endpoint):
+                response = self.client.get(f"/{STAC_BASE_V}/{endpoint}?limit=1")
+                json_data = response.json()
+                self.assertEqual(
+                    200, response.status_code, msg=get_http_error_description(json_data)
+                )
 
-        response = self.client.get(f"/{STAC_BASE_V}/collections?limit=1")
-        json_data = response.json()
-        self.assertEqual(200, response.status_code, msg=get_http_error_description(json_data))
-
-        # Check next link
-        next_link = get_link(json_data['links'], 'next')
-        self.assertIsNotNone(next_link, msg='Pagination next link missing')
-        self.assertTrue(isinstance(next_link['href'], str), msg='href is not a string')
-        self.assertTrue(
-            next_link['href'].startswith('http://testserver/api/stac/v0.9/collections?cursor='),
-            msg='Invalid href link pagination string'
-        )
+                # Check next link
+                next_link = get_link(json_data['links'], 'next')
+                self.assertIsNotNone(next_link, msg='Pagination next link missing')
+                self.assertTrue(isinstance(next_link['href'], str), msg='href is not a string')
+                self.assertTrue(
+                    next_link['href'].
+                    startswith(f'http://testserver/api/stac/v0.9/{endpoint}?cursor='),
+                    msg='Invalid href link pagination string'
+                )
```

Comment:
Not really relevant for this PR, but when in a call with Isabelle about the e2e tests, we decided to implement a `_check_link()` and pass the response and the `link_rel` to it.
So the lines after the comments `#check next link` and `#check previous link` (each called twice here) can be handled by that new function.

Reply:
good point I'll do it.

Reply:
Done see commit https://github.com/geoadmin/service-stac/pull/251/commits/fbff8f3b6855a48a662ff6911772f26e66489ae3